src/overlays/gamestates/ovl_select/z_select.c

```diff
@@ -1,25 +1,633 @@
+
+        if (this->currentScene == ((this->unk_20C + this->count) % this->count)) {
+            this->unk_20C = (--this->unk_20C + this->count) % this->count;
+        }
+    }
+
+    this->currentScene = (this->currentScene + this->count) % this->count;
+    this->unk_20C = (this->unk_20C + this->count) % this->count;
+
+    dREG(80) = this->currentScene;
+    dREG(81) = this->unk_20C;
+    dREG(82) = this->unk_1DC;
+
+    if (this->unk_224 != 0) {
+        this->unk_224--;
+    }
+
+    if (this->unk_224 == 0) {
+        this->unk_22C = 0;
+    }
+
+    if (this->unk_228 != 0) {
+        this->unk_228--;
+    }
+
+    if (this->unk_228 == 0) {
+        this->unk_230 = 0;
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/gamestates/ovl_select/Select_UpdateMenu.s")
+#endif
+
+void Select_PrintMenu(SelectContext* this, GfxPrint* printer) {
+    s32 scene;
+    s32 i;
+    char* name;
+
+    GfxPrint_SetColor(printer, 255, 155, 150, 255);
+    GfxPrint_SetPos(printer, 12, 2);
+    GfxPrint_Printf(printer, "ZELDA MAP SELECT");
+    GfxPrint_SetColor(printer, 255, 255, 255, 255);
+
+    for (i = 0; i < 20; i++) {
+        GfxPrint_SetPos(printer, 9, i + 4);
+
+        scene = ((this->unk_20C + i) + this->count) % this->count;
+        if (scene == this->currentScene) {
+            GfxPrint_SetColor(printer, 255, 20, 20, 255);
+        } else {
+            GfxPrint_SetColor(printer, 200, 200, 55, 255);
+        }
+
+        name = this->scenes[scene].name;
+        if (name == NULL) {
+            name = "**Null**";
+        }
+
+        GfxPrint_Printf(printer, "%s", name);
+    };
+
+    GfxPrint_SetColor(printer, 155, 55, 150, 255);
+    GfxPrint_SetPos(printer, 20, 26);
+    GfxPrint_Printf(printer, "OPT=%d", this->opt);
+}
+
+static char* sLoadingMessages[] = {
+    "\x8Dｼﾊﾞﾗｸｵﾏﾁｸﾀﾞｻｲ", // "Please wait a minute"
+    "\x8Dﾁｮｯﾄ ﾏｯﾃﾈ",     // "Hold on a sec"
+    "\x8Cｳｪｲﾄ ｱ ﾓｰﾒﾝﾄ",  // "Wait a moment"
+    "\x8Cﾛｰﾄﾞ\x8Dﾁｭｳ",   // "Loading"
+    "\x8Dﾅｳ ﾜｰｷﾝｸﾞ",     // "Now working"
+    "\x8Dｲﾏ ﾂｸｯﾃﾏｽ",     // "Now creating"
+    "\x8Dｺｼｮｳｼﾞｬﾅｲﾖ",    // "It's not broken"
+    "\x8Cｺｰﾋｰ ﾌﾞﾚｲｸ",    // "Coffee Break"
+    "\x8C"
+    "Bﾒﾝｦｾｯﾄｼﾃｸﾀﾞｻｲ",                      // "Please set B side"
+    "\x8Dｼﾞｯﾄ\x8Cｶﾞﾏﾝ\x8Dﾉ\x8Cｺ\x8Dﾃﾞｱｯﾀ", // "Be patient, now"
+    "\x8Dｲﾏｼﾊﾞﾗｸｵﾏﾁｸﾀﾞｻｲ",                 // "Please wait just a minute"
+    "\x8Dｱﾜﾃﾅｲｱﾜﾃﾅｲ｡ﾋﾄﾔｽﾐﾋﾄﾔｽﾐ｡",          // "Don't worry, don't worry. Take a break, take a break"
+};
+
+void Select_PrintLoadingMessage(SelectContext* this, GfxPrint* printer) {
+    s32 randomMsg;
+
+    GfxPrint_SetPos(printer, 0xA, 0xF);
+    GfxPrint_SetColor(printer, 255, 255, 255, 255);
+    randomMsg = Math_Rand_ZeroOne() * ARRAY_COUNT(sLoadingMessages);
+    GfxPrint_Printf(printer, "%s", sLoadingMessages[randomMsg]);
+}
+
+static char* sAgeLabels[] = {
+    "\x8D"
+    "17(ﾜｶﾓﾉ)", // "17(young)"
+    "\x8D"
+    "5(ﾜｶｽｷﾞ)", // "5(very young)"
```

Comment:
```suggestion
    "\x8D\17(ﾜｶﾓﾉ)", // "17(young)"
    "\x8D\5(ﾜｶｽｷﾞ)", // "5(very young)"
```

Reply:
this doesnt work, or the one above

Reply:
actually it works for the one above but not this one?? lol